### PR TITLE
Windows Update

### PR DESCRIPTION
**Pull request by uraniummeltdown**
windows

Fixes https://github.com/ParadiseSS13/Paradise/issues/8890

window construction and the special fulltile diagonal have defines
added proc to check for windoor(assembly)/window
added var/fulltile to windows, no longer uses a proc
added var/cancolor to windows to determine whether they should get colored or not
added clockwork windoors and windoor ratvar_act()
added titanium glass and plastitanium glass. Titanium + Glass in ORM for titanium glass, Titanium + Plasma + Glass for plastitanium glass
slight nerf to fireaxe, it does dmg to windows and grilles instead of insta delete. Windows converted to new obj dmg system
glass stacks now use recipes instead of a menu
fullwindow.dm deleted and moved into window.dm
full windows now have crack overlays when they reach certain damage percentages
windows have deconstruction hints
windows can be repaired with help intent welder
very minor buffs to window health

:cl:
add: You can smelt titanium and glass together to form titanium glass for building shuttle windows
add: You can smelt titanium, plasma and glass together to form plastitanium glass for building plastitanium windows
add: Fulltile windows now smooth, windows crack as they get damaged and can be repaired with help intent welder
add: Windows have deconstruction hints and show whether they can be rotated or not
fix: Plasma glass no longer gets auto-colored
fix: RCDs can deconstruct airlocks again, they have no force now though
tweak: Wielded fireaxe does a lot of damage to windows and grilles instead of just deleting them
tweak: Glass stacks now use stack recipes instead of a custom menu
/:cl:

### CODE REVIEW

**Comment by SkeletalElite** [comment]
Energy Axe charges up when breaking grilles doesn't it? Will this break that?

**Comment by uraniummeltdown** [comment]
Nope, energy axes work fine

**Comment by taukausanake** [comment]
I forget it Nar-sie cult have windows or not. If not you think those could get added?

**Comment by uraniummeltdown** [comment]
I can't sprite new cult windows because I can't sprite, but currently narsie_act() will color windows crimson/blood-red

**Comment by matt81093** [comment]
This sounds awesome! Looking forward to these additions.

**Comment by uraniummeltdown** [comment]
@Fox-McCloud fixed merge conflicts

File: code/modules/mining/equipment_locker.dm [hunk 1]
```diff
@@ -197,6 +197,8 @@
 		dat += text("Plastitanium Alloy (Titanium + Plasma): <A href='?src=[UID()];alloytype1=/obj/item/stack/sheet/mineral/titanium;alloytype2=/obj/item/stack/sheet/mineral/plasma;alloytypeout=/obj/item/stack/sheet/mineral/plastitanium'>Smelt</A><BR>")
 	if(glassstack && plasmastack && min(glassstack.amount, plasmastack.amount))
 		dat += text("Plasma Glass (Glass + Plasma): <A href='?src=[UID()];alloytype1=/obj/item/stack/sheet/glass;alloytype2=/obj/item/stack/sheet/mineral/plasma;alloytypeout=/obj/item/stack/sheet/plasmaglass'>Smelt</A><BR>")
+	if(glassstack && titaniumstack && min(glassstack.amount, titaniumstack.amount))
+		dat += text("Titanium Glass (Titanium + Glass): <A href='?src=[UID()];alloytype1=/obj/item/stack/sheet/glass;alloytype2=/obj/item/stack/sheet/mineral/titanium;alloytypeout=/obj/item/stack/sheet/titaniumglass'>Smelt</A><BR>")
```
Comment (Fox-McCloud) [inline code/modules/mining/equipment_locker.dm]: Going to hold off on merging this until the ORM refactor goes through.

On the plus side, by waiting, to implement new alloys, you only have to add them to here: **https://github.com/ParadiseSS13/Paradise/blob/master/code/modules/research/designs/smelting_designs.dm**

Comment (uraniummeltdown) [reply code/modules/mining/equipment_locker.dm]: @Fox-McCloud Thanks for doing the refactor, makes this a whole lot simpler

File: code/game/machinery/doors/door.dm [hunk 2]
```diff
@@ -112,7 +112,7 @@
 
 /obj/machinery/door/Move(new_loc, new_dir)
 	var/turf/T = loc
-	. = ..()
```
Comment (tigercat2000) [inline code/game/machinery/doors/door.dm]: What's with this?

**Pull request merged**